cg/services/order_validation_service/rules/case_sample/utils.py

```diff
@@ -22,3 +51,252 @@ def get_invalid_panels(panels: list[str], store: Store) -> list[str]:
+    return father and father.sex != Sex.MALE
+
+
+def create_father_sex_error(case_index: int, sample_index: int) -> InvalidFatherSexError:
+    return InvalidFatherSexError(case_index=case_index, sample_index=sample_index)
+
+
+def get_father_case_errors(
+    case: CaseContainingRelatives,
+    case_index: int,
+) -> list[FatherNotInCaseError]:
+    errors: list[FatherNotInCaseError] = []
+    children: list[tuple[SampleWithRelatives, int]] = case.get_samples_with_father()
+    for child, child_index in children:
+        father: SampleWithRelatives | None = case.get_sample(child.father)
+        if not father:
+            error: FatherNotInCaseError = create_father_case_error(
+                case_index=case_index,
+                sample_index=child_index,
+            )
+            errors.append(error)
+    return errors
+
+
+def get_mother_sex_errors(
+    case: CaseContainingRelatives,
+    case_index: int,
+) -> list[InvalidMotherSexError]:
+    errors: list[InvalidMotherSexError] = []
+    children: list[tuple[SampleWithRelatives, int]] = case.get_samples_with_mother()
+    for child, child_index in children:
+        if is_mother_sex_invalid(child=child, case=case):
+            error: InvalidMotherSexError = create_mother_sex_error(
+                case_index=case_index,
+                sample_index=child_index,
+            )
+            errors.append(error)
+    return errors
+
+
+def get_mother_case_errors(
+    case: CaseContainingRelatives,
+    case_index: int,
+) -> list[MotherNotInCaseError]:
+    errors: list[MotherNotInCaseError] = []
+    children: list[tuple[SampleWithRelatives, int]] = case.get_samples_with_mother()
+    for child, child_index in children:
+        mother: SampleWithRelatives | None = case.get_sample(child.mother)
+        if not mother:
+            error: MotherNotInCaseError = create_mother_case_error(
+                case_index=case_index, sample_index=child_index
+            )
+            errors.append(error)
+    return errors
+
+
+def create_father_case_error(case_index: int, sample_index: int) -> FatherNotInCaseError:
+    return FatherNotInCaseError(case_index=case_index, sample_index=sample_index)
+
+
+def create_mother_case_error(case_index: int, sample_index: int) -> MotherNotInCaseError:
+    return MotherNotInCaseError(case_index=case_index, sample_index=sample_index)
+
+
+def is_mother_sex_invalid(child: SampleWithRelatives, case: CaseContainingRelatives) -> bool:
+    mother: SampleWithRelatives | None = case.get_sample(child.mother)
+    return mother and mother.sex != Sex.FEMALE
+
+
+def create_mother_sex_error(case_index: int, sample_index: int) -> InvalidMotherSexError:
+    return InvalidMotherSexError(case_index=case_index, sample_index=sample_index)
+
+
+def validate_subject_ids_in_case(
+    case: CaseContainingRelatives, case_index: int
+) -> list[SubjectIdSameAsCaseNameError]:
+    errors: list[SubjectIdSameAsCaseNameError] = []
```

Comment:
Again, nothing to do with relatives and will probably need another typing but keep it for now.